packages/xo-web/src/xo-app/sr/tab-disks.js

```diff
@@ -89,13 +52,87 @@ const COLUMNS = [
     itemRenderer: vdi => formatSize(vdi.size),
     sortCriteria: vdi => vdi.size,
   },
+  {
+    name: _('vdiVms'),
+    component: connectStore(() => {
+      const getVbds = createGetObjectsOfType('VBD').pick(
+        (_, props) => props.item.$VBDs
+      )
+      const getVmIds = createSelector(getVbds, vbds => map(vbds, 'VM'))
+      const getVms = createGetObjectsOfType('VM').pick(getVmIds)
+      const getVmSnapshots = createGetObjectsOfType('VM-snapshot').pick(
```

Comment:
Actually, you could simply test if `item.type` is either `VDI` or `VDI-snapshot` and only get the objects you really need: `VM`s or `VM-snapshot`s.